### ISSUE TITLE
DotNet version does not support escaped strings in responses

### ISSUE THREAD

**Issue by seckin92** [Esri/resource-proxy#474]
When used against LocatorHub (our version was 10.2.2), the LocatorHub responses contain escaped forward slashes (`\/`), e.g.

    {
        "tokenServicesUrl": "https:\/\/my.gis.server\/locatorhub\/arcgis\/tokens"
    }
This causes the requests to end up with the following exception: 

    UriFormatException: Invalid URI: The hostname could not be parsed.

**Comment by bsvensson** [comment]
Hi @seckin92 This sounds more like a problem with the LocatorHub.  I would also report that as a bug to them.  It's possible that those backslashes might break other things as well.

You found this issue with the DotNet proxy.  Any chance you also tried (or could try) the php & java proxies?

**Comment by seckin92** [comment]
Hi @bsvensson. That was my initial assumption as well, but I weren't able to raise this with ESRI as the LocatorHub is deployed in a client's environment. I advised the client to raise this, I'll update this thread if / when there are results.

Regarding the Java and PHP proxies, I haven't tried the proxies themselves, but I snipped the JSON parsing bits and tested them with the escaped value.

I tested the Java version with the following code, and from the output, I would say the Java version also has this issue.
```java
	private static String getJsonValue(String text, String key) {
        // _log(Level.FINE, "JSON Response: " + text);
        int i = text.indexOf(key);
        String value = "";
        if (i > -1) {
            value = text.substring(text.indexOf(':', i) + 1).trim();
            value = (value.length() > 0 && value.charAt(0) == '"') ?
                    value.substring(1, value.indexOf('"', 1)) :
                    value.substring(0, Math.max(0, Math.min(Math.min(value.indexOf(","), value.indexOf("]")), value.indexOf("}"))));
        }
        // _log(Level.FINE, "Extracted Value: " + value);
        return value;
    }
    
	public static void main (String[] args) throws java.lang.Exception
	{
		String json = "{"
		+    "\"tokenServicesUrl\": \"https:\\/\\/my.gis.server\\/locatorhub\\/arcgis\\/tokens\""
		+ "}";
		
		String tokenServicesUrl = getJsonValue(json, "tokenServicesUrl");
		
		System.out.println("json = " + json);
		System.out.println("tokenServicesUrl = " + tokenServicesUrl);

        // Output
        //
        // json = {"tokenServicesUrl": "https:\/\/my.gis.server\/locatorhub\/arcgis\/tokens"}
        // tokenServicesUrl = https:\/\/my.gis.server\/locatorhub\/arcgis\/tokens
	}
```

The PHP version uses `json_decode`, I tested this with the following code, and the output looks fine.
```php
<?php

$json = <<<_EOT
{
    "tokenServicesUrl": "https:\/\/my.gis.server\/locatorhub\/arcgis\/tokens"
}
_EOT;

$parsed = json_decode($json, true);
echo "\$json = ";
var_dump($json);

echo "\$parsed = ";
var_dump($parsed);

// Output:
// 
// $json = string(81) "{
//     "tokenServicesUrl": "https:\/\/my.gis.server\/locatorhub\/arcgis\/tokens"
// }"
// $parsed = array(1) {
//   ["tokenServicesUrl"]=>
//   string(46) "https://my.gis.server/locatorhub/arcgis/tokens"
// }
```

**Comment by bsvensson** [comment]
Thanks for testing,  and contributing. I've merged the php PR.

**Comment by seckin92** [comment]
Addressed by #475
